lib/src/async/retry.dart

```diff
@@ -21,13 +21,22 @@ const _defaultTimeout = const Duration(seconds: 60);
 /// reached.
 ///
 /// Waits [interval] to perform the retry after a failed attempt.
+///
+/// If the task fails to complete once within the timeout it will resolve as an
+/// error with a TimeoutException. If the task completes with an error at least
+/// once before the timeout and never succeeds it will resolve as an error with
+/// the last exception from a task run that did not time out.
 Future/*<T>*/ retry/*<T>*/(Future/*<T>*/ task(),
     {Duration interval: _defaultInterval,
     Duration timeout: _defaultTimeout}) async {
   var end = new DateTime.now().add(timeout);
+  dynamic lastCaught;
```

Comment:
dynamic is more future-proof if we decide to turn on --no-implicit-dynamic on the analyzer (which I hope we do). To that end I also added a generic argument to the Future.delayed below. I wasn't concerned with --no-implicit-dynamic in the tests but I could update those too.